atlasdb-config/src/main/java/com/palantir/atlasdb/factory/InitialisingTransactionManager.java

```diff
@@ -0,0 +1,630 @@
+import com.palantir.atlasdb.transaction.impl.ConflictDetectionManagers;
+import com.palantir.atlasdb.transaction.impl.RawTransaction;
+import com.palantir.atlasdb.transaction.impl.SerializableTransactionManager;
+import com.palantir.atlasdb.transaction.impl.SerializableTransactionManagerImpl;
+import com.palantir.atlasdb.transaction.impl.SweepStrategyManager;
+import com.palantir.atlasdb.transaction.impl.SweepStrategyManagers;
+import com.palantir.atlasdb.transaction.impl.TransactionTables;
+import com.palantir.atlasdb.transaction.service.TransactionService;
+import com.palantir.atlasdb.transaction.service.TransactionServices;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.atlasdb.util.JavaSuppliers;
+import com.palantir.leader.LeaderElectionService;
+import com.palantir.leader.PingableLeader;
+import com.palantir.leader.proxy.AwaitingLeadershipProxy;
+import com.palantir.lock.HeldLocksToken;
+import com.palantir.lock.InitialisingRemoteLockService;
+import com.palantir.lock.LockClient;
+import com.palantir.lock.LockRefreshToken;
+import com.palantir.lock.LockRequest;
+import com.palantir.lock.LockServerOptions;
+import com.palantir.lock.RemoteLockService;
+import com.palantir.lock.SimpleTimeDuration;
+import com.palantir.lock.client.LockRefreshingRemoteLockService;
+import com.palantir.lock.impl.LockServiceImpl;
+import com.palantir.timestamp.InitialisingTimestampService;
+import com.palantir.timestamp.TimestampService;
+import com.palantir.timestamp.TimestampStoreInvalidator;
+
+public class InitialisingTransactionManager extends ForwardingObject implements SerializableTransactionManager {
+
+    public static final Logger log = LoggerFactory.getLogger(InitialisingTransactionManager.class);
+    private static final int LOGGING_INTERVAL = 60;
+    private static final LockClient LOCK_CLIENT = LockClient.of("atlas instance");
+
+    private volatile SerializableTransactionManagerImpl delegate = null;
+    private Leaders.LocalPaxosServices localPaxosServices = null;
+    private final InitialisingRemoteLockService remoteLockService = InitialisingRemoteLockService.create();
+    private final InitialisingTimestampService timestampService = InitialisingTimestampService.createUninitialised();
+    private final InitialisingPersistentLockService persistentLockService = InitialisingPersistentLockService.create();
+    private final InitialisingSweeperService sweeperService = InitialisingSweeperService.create();
+
+    private final ServiceDiscoveringAtlasSupplier atlasFactory;
+    private final AtlasDbConfig config;
+    private final java.util.function.Supplier<java.util.Optional<AtlasDbRuntimeConfig>> runtimeConfigSupplier;
+    private final Set<Schema> schemas;
+    private final TransactionManagers.Environment env;
+    private final LockServerOptions lockServerOptions;
+    private final boolean allowHiddenTableAccess;
+    private final String userAgent;
+
+    static Consumer<Runnable> runAsync = task -> {
```

Comment:
This might lead to multiple threads trying to initialize the KVS. Should we use the single-threaded ScheduledThreadPoolExecutor here? @nziebart for thoughts?